samples/features/azure-arc-data-controller/deployment/kubeadm/ubuntu-single-node-vm/setup-controller.sh

```diff
@@ -0,0 +1,299 @@
+
+# Setup kubernetes cluster including remove taint on master.
+#
+echo ""
+echo "#############################################################################"
+echo "Starting to setup Kubernetes master..." 
+
+# Initialize a kubernetes cluster on the current node.
+#
+sudo kubeadm init --pod-network-cidr=10.244.0.0/16 --kubernetes-version=$KUBE_VERSION
+
+mkdir -p $HOME/.kube
+mkdir -p /home/$SUDO_USER/.kube
+
+sudo cp -f /etc/kubernetes/admin.conf $HOME/.kube/config
+sudo chown $(id -u $SUDO_USER):$(id -g $SUDO_USER) $HOME/.kube/config
+
+# To enable a single node cluster remove the taint that limits the first node to master only service.
+#
+master_node=`kubectl get nodes --no-headers=true --output=custom-columns=NAME:.metadata.name`
+kubectl taint nodes ${master_node} node-role.kubernetes.io/master:NoSchedule-
+
+# Local storage provisioning.
+#
+kubectl apply -f https://raw.githubusercontent.com/ananto-msft/sql-server-samples/$GITHUB_AADATACONTROLLER_BRANCH/samples/features/azure-arc-data-controller/deployment/kubeadm/ubuntu/local-storage-provisioner.yaml
+
+# Install the software defined network.
+#
+kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml
+
+# helm init
+#
+kubectl apply -f https://raw.githubusercontent.com/ananto-msft/sql-server-samples/$GITHUB_AADATACONTROLLER_BRANCH/samples/features/azure-arc-data-controller/deployment/kubeadm/ubuntu/rbac.yaml
+
+# Verify that the cluster is ready to be used.
+#
+echo "Verifying that the cluster is ready for use..."
+while true ; do
+
+    if [[ "$TIMEOUT" -le 0 ]]; then
+        echo "Cluster node failed to reach the 'Ready' state. Kubeadm setup failed."
+        exit 1
+    fi
+
+    status=`kubectl get nodes --no-headers=true | awk '{print $2}'`
+
+    if [ "$status" == "Ready" ]; then
+        break
+    fi
+
+    sleep "$RETRY_INTERVAL"
+
+    TIMEOUT=$(($TIMEOUT-$RETRY_INTERVAL))
+
+    echo "Cluster not ready. Retrying..."
+done
+
+
+# Install the dashboard for Kubernetes.
+#
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/dashboard/v1.10.1/src/deploy/recommended/kubernetes-dashboard.yaml
+
+kubectl create clusterrolebinding kubernetes-dashboard --clusterrole=cluster-admin --serviceaccount=kube-system:kubernetes-dashboard
+echo "Kubernetes master setup done."
+
+# Deploy azdata Azure Arc Data Cotnroller create cluster.
+#
+echo ""
+echo "############################################################################"
+echo "Starting to deploy azdata cluster..." 
+
+# Command to create cluster for single node cluster.
+#
+azdata control create -c tina-kubeadm-private-preview --accept-eula $ACCEPT_EULA
+echo "Azure Arc Data Controller cluster created." 
+
+# Setting context to cluster.
+#
+kubectl config set-context --current --namespace $CLUSTER_NAME
+
+# Login and get endpoint list for the cluster.
+#
+azdata login -n $CLUSTER_NAME
+azdata bdc endpoint list --output table
```

Comment:
I don't think this is applicable in Arc world. You are not deploying BDC so this will fail.

Reply:
Doesn't fail but I'll remove it.